Codigo/SistemaCombate.bas

```diff
@@ -1833,6 +1833,45 @@ Public Function PuedeAtacarNPC(ByVal AttackerIndex As Integer, ByVal NpcIndex As
                         Exit Function
                 End Select
             End If
+            
+            
+            If NpcList(NpcIndex).flags.AttackedBy <> "" Then
+                Dim AttackedBy As t_UserReference
+                Dim AttackedByIndex As Integer
+                Dim Lider As t_User
+                Dim a As Integer
+                
+
+                    AttackedBy = NameIndex(NpcList(NpcIndex).flags.AttackedBy)
+                    If IsValidUserRef(AttackedBy) Then
+                        AttackedByIndex = AttackedBy.ArrayIndex
+                        If AttackedByIndex <> attackerIndex Then
+                            If UserList(AttackedByIndex).flags.AttackedNpc = NpcIndex And UserList(AttackedByIndex).flags.Muerto = 0 And Status(AttackedByIndex) = Ciudadano And (UserList(attackerIndex).GuildIndex = 0 Or UserList(attackerIndex).GuildIndex <> UserList(AttackedByIndex).GuildIndex) Then
+                                
+                                If UserList(attackerIndex).Grupo.EnGrupo Then
+                                    Lider = UserList(UserList(attackerIndex).Grupo.Lider.ArrayIndex)
```

Comment:
no need to loop all group members, you can test the groupo.id for both new attacker and the currentowner